contracts/cw721-non-transferable/src/msg.rs

```diff
@@ -0,0 +1,11 @@
+use cosmwasm_schema::cw_serde;
+
+use cosmwasm_std::Addr;
+
+#[cw_serde]
+pub struct InstantiateMsg {
+    pub admin: Option<Addr>,
```

Comment:
`admin` should be `Option<String>`. The instantiation function should validate the address, e.g.

```rust
// `msg.admin` is of `Option<String>` type  
let admin_addr: Addr = msg
    .admin
    .map(|s| deps.api.addr_validate(&s))
    .transpose()?;
```

Reply:
if you're feeling really fancy, you can even get the value inside the option as a reference to avoid a move:

```rust
let admin = msg
    .admin  // Option<String>
    .as_deref()  // Option<&str>
    .map(|h| deps.api.addr_validate(h))  // Option<StdResult<Addr>>
    .transpose()?;  // StdResult<Option<Addr>>
```

Reply:
fixed it, ty both